apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Trace/CloudTraceExtension.cs

```diff
@@ -142,8 +140,7 @@ public static void UseGoogleTrace(this IApplicationBuilder app)
             services.AddSingleton<IManagedTracer>(CreateManagedTracer);
 
             services.AddSingleton(CreateTraceHeaderPropagatingHandler);
-            services.AddSingleton(new ShouldTraceRequest(traceOverridePredicate));
-            services.AddSingleton(traceIdFactory);
+            services.AddSingleton(new ShouldTraceRequest(traceFallbackPredicate));
```

Comment:
I think I was going to send you a PR to see if I could easily remove that, wasn't I?